pkg/connect/client/datastore/v1alpha1/fake/fake_test.go

```diff
@@ -0,0 +1,300 @@
+	require.NoError(t, err)
+	assert.Equal(t, int32(5), countResp.Count)
+}
+
+func TestSetAndGetNodeSelectors(t *testing.T) {
+	_, client := setupTest()
+
+	spiffeID := "spiffe://example.org/node"
+	selectors := []*datastorev1alpha1.Selector{
+		{Type: "type1", Value: "value1"},
+		{Type: "type2", Value: "value2"},
+	}
+
+	node := &datastorev1alpha1.AttestedNode{
+		SpiffeId: spiffeID,
+	}
+	_, err := client.CreateAttestedNode(context.Background(), &datastorev1alpha1.CreateAttestedNodeRequest{
+		Node: node,
+	})
+	require.NoError(t, err)
+	_, err = client.SetNodeSelectors(context.Background(), &datastorev1alpha1.SetNodeSelectorsRequest{
+		SpiffeId:  spiffeID,
+		Selectors: selectors,
+	})
+	require.NoError(t, err)
+
+	getResp, err := client.GetNodeSelectors(context.Background(), &datastorev1alpha1.GetNodeSelectorsRequest{
+		SpiffeId: spiffeID,
+	})
+	require.NoError(t, err)
+	require.Len(t, getResp.Selectors, 2)
+	assert.Equal(t, selectors, getResp.Selectors)
+}
+func TestListAttestedNodes(t *testing.T) {
+	fake, client := setupTest()
+
+	// Clear existing nodes
+	fake.AttestedNodes = make(map[string]*datastorev1alpha1.AttestedNode)
+
+	// Create test nodes
+	nodes := make([]*datastorev1alpha1.AttestedNode, 3)
+	for i := 0; i < 3; i++ {
+		node := &datastorev1alpha1.AttestedNode{
+			CertSerialNumber: "list-test-" + strconv.Itoa(i),
+			CertNotAfter:     int64(i * 1000),
+			SpiffeId:         "spiffe://example.org/list-node-" + strconv.Itoa(i),
+		}
+		nodes[i] = node
+		_, err := client.CreateAttestedNode(context.Background(), &datastorev1alpha1.CreateAttestedNodeRequest{
+			Node: node,
+		})
+		require.NoError(t, err)
+	}
+
+	listResp, err := client.ListAttestedNodes(context.Background(), &datastorev1alpha1.ListAttestedNodesRequest{})
+	require.NoError(t, err)
+	require.NotNil(t, listResp)
+	assert.Len(t, listResp.Nodes, 3)
+	
+	// Verify all nodes are returned
+	for _, expectedNode := range nodes {
+		found := false
+		for _, actualNode := range listResp.Nodes {
+			if actualNode.SpiffeId == expectedNode.SpiffeId {
+				found = true
+				assert.Equal(t, expectedNode.CertSerialNumber, actualNode.CertSerialNumber)
+				assert.Equal(t, expectedNode.CertNotAfter, actualNode.CertNotAfter)
+				break
+			}
+		}
+		assert.True(t, found, "Node not found in list response: %s", expectedNode.SpiffeId)
+	}
+}
+
+func TestListNodeSelectors(t *testing.T) {
+	fake, client := setupTest()
+
+	// Clear existing nodes
+	fake.AttestedNodes = make(map[string]*datastorev1alpha1.AttestedNode)
+
+	// Create test nodes with selectors
+	nodeCount := 3
+	for i := 0; i < nodeCount; i++ {
+		spiffeID := "spiffe://example.org/selector-node-" + strconv.Itoa(i)
+		node := &datastorev1alpha1.AttestedNode{
+			CertSerialNumber: "selector-test-" + strconv.Itoa(i),
+			SpiffeId:         spiffeID,
+			Selectors: []*datastorev1alpha1.Selector{
+				{Type: "type" + strconv.Itoa(i), Value: "value" + strconv.Itoa(i)},
+				{Type: "common", Value: "shared"},
+			},
```

Comment:
The SPIRE SQL datastore doesn't create the selectors when it creates a node. They get created separately using SetNodeSelectors.